source/base/utilities.cc

```diff
@@ -904,79 +906,36 @@ namespace Utilities
     return out;
   }
 
-  std::vector<unsigned long long int>
-  reverse_permutation(const std::vector<unsigned long long int> &permutation)
-  {
-    const unsigned long long int n = permutation.size();
 
-    std::vector<unsigned long long int> out(n);
-    for (unsigned long long int i = 0; i < n; ++i)
-      out[i] = n - 1 - permutation[i];
 
-    return out;
+  std::vector<unsigned int>
+  reverse_permutation(const std::vector<unsigned int> &permutation)
+  {
+    return reverse_permutation<unsigned int>(permutation);
```

Comment:
do we even still need this function then?

Reply:
@kronbichler wanted to avoid a link-time error this way. As I suggested before, I would be fine with moving the declaration and the definition of the templated version to the header file and remove these two overloads. I don't feel strongly about it, though.